lib/puppet/type/file/source.rb

```diff
@@ -72,7 +72,7 @@ module Puppet
           file { '/etc/nfs.conf':
             source => [
               "puppet:///modules/nfs/conf.${host}",
-              "puppet:///modules/nfs/conf.${operatingsystem}",
+              "puppet:///modules/nfs/conf.${os['name']}",
```

Comment:
This one is a bit tricky because there's double '$':

```suggestion
              "puppet:///modules/nfs/conf.${$facts.dig('os', 'name')}"
```

Reply:
oh nevermind, this is fine: 

```
❯ cat manifest.pp 
notice("puppet:///modules/nfs/conf.${os['name']}")

❯ bx puppet apply manifest.pp
Notice: Scope(Class[main]): puppet:///modules/nfs/conf.Ubuntu
```